File: fsharp/lucians-luscious-lasagna/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

let expectedMinutesInOven = 40

let remainingMinutesInOven elapsed = expectedMinutesInOven - elapsed

let preparationTimeInMinutes layer = layer * 2

let elapsedTimeInMinutes layer elapsed =
    preparationTimeInMinutes layer + elapsed

File: csharp/lucians-luscious-lasagna/LuciansLusciousLasagna.cs
class Lasagna
{
    public int ExpectedMinutesInOven() => 40;

    public int RemainingMinutesInOven(int actualMinutes) => ExpectedMinutesInOven() - actualMinutes;

    public int PreparationTimeInMinutes(int layers) => 2 * layers;

    public int ElapsedTimeInMinutes(int layers, int ovenMinutes)
    {
        var prepTime = PreparationTimeInMinutes(layers);
        return prepTime + ovenMinutes;
    }
}
